presto-main/src/main/java/com/facebook/presto/operator/scalar/annotations/ScalarImplementation.java

```diff
@@ -253,9 +254,26 @@ private TypeImplementationDependency(String signature)
         }
 
         @Override
-        public Type resolve(Map<String, Type> types, TypeManager typeManager, FunctionRegistry functionRegistry)
+        public Type resolve(BoundVariables boundVariables, TypeManager typeManager, FunctionRegistry functionRegistry)
         {
-            return typeManager.getType(SignatureBinder.bindVariables(signature, new BoundVariables(types, ImmutableMap.of())));
+            return typeManager.getType(SignatureBinder.bindVariables(signature, boundVariables));
+        }
+    }
+
+    private static final class LiteralImplementationDependency
+            implements ImplementationDependency
+    {
+        private final String literal;
```

Comment:
Lets name this `variable` or `literalName` it sounds like it's the value of the literal the way it is now